### PR TITLE
Suzuka Doc Comments

### PR DESCRIPTION
**Pull request by BenDavis36**
# Summary
Creation of doc comments for Suzuka

# Changelog
Adds comments for `struct SuzukaPartialNode<T>` and associated fields.
Adds comments for `pub struct Config`.

### CODE REVIEW

**Review by 0xmovses — CHANGES_REQUESTED** [review]

File: networks/suzuka/suzuka-full-node/src/partial.rs [hunk 1]
```diff
@@ -24,11 +24,36 @@ use std::future::Future;
 use std::sync::Arc;
 use std::time::Duration;
 
+/// Suzuka partial node has the ability to
```
Comment (mzabaluev) [inline networks/suzuka/suzuka-full-node/src/partial.rs:27]: This needs to be completed.

File: networks/suzuka/suzuka-full-node/src/partial.rs [hunk 2]
```diff
@@ -24,11 +24,36 @@ use std::future::Future;
...
 pub struct SuzukaPartialNode<T> {
+	/// The executor responsible for executing transactions and blocks.
+	///
+	/// 'T' must implement the `SuzukaExecutor` trait, allowing different
+	/// types of executors to be used with the `SuzukaPartialNode`.
```
Comment (mzabaluev) [inline networks/suzuka/suzuka-full-node/src/partial.rs:32]: Private members do not really need doc comments, as they won't be rendered in the API documents. A plain one-line comment describing the purpose should be enough.

Comment (0xmovses) [reply networks/suzuka/suzuka-full-node/src/partial.rs:32]: There are tonnes of private members in aptos core that are doc commented. This is for IDE support more than rust crate docs I suppose, for eg.

```rust
/// See [AptosDB::get_transaction_by_version].
        ///
        /// [AptosDB::get_transaction_by_version]: ../aptosdb/struct.AptosDB.html#method.get_transaction_by_version
        fn get_transaction_by_version(
            &self,
            version: Version,
            ledger_version: Version,
            fetch_events: bool,
        ) -> Result<TransactionWithProof>;
        
        ```

Comment (mzabaluev) [reply networks/suzuka/suzuka-full-node/src/partial.rs:32]: I just checked and you are right, this gives better quality of life in the IDE.

**Pull request closed without merging**